Source/WebCore/html/HTMLCanvasElement.cpp

```diff
@@ -246,17 +246,16 @@ ExceptionOr<std::optional<RenderingContext>> HTMLCanvasElement::getContext(JSC::
             auto version = toWebGLVersion(contextId);
             if ((version == WebGLVersion::WebGL1) != m_context->isWebGL1())
                 return std::optional<RenderingContext> { std::nullopt };
-            if (is<WebGLRenderingContext>(*m_context))
-                return std::optional<RenderingContext> { RefPtr<WebGLRenderingContext> { &downcast<WebGLRenderingContext>(*m_context) } };
-            ASSERT(is<WebGL2RenderingContext>(*m_context));
-            return std::optional<RenderingContext> { RefPtr<WebGL2RenderingContext> { &downcast<WebGL2RenderingContext>(*m_context) } };
+            if (auto* context = dynamicDowncast<WebGLRenderingContext>(*m_context))
+                return std::optional<RenderingContext> { RefPtr { context } };
+            return std::optional<RenderingContext> { RefPtr { &downcast<WebGL2RenderingContext>(*m_context) } };
         }
 #endif
 
-        if (m_context->isWebGPU()) {
+        if (auto* context = dynamicDowncast<GPUCanvasContext>(m_context.get())) {
             if (!isWebGPUType(contextId))
```

Comment:
existing, but `isWebGPUType` test could be moved up. potentially remove one check.

Reply:
Ok, will look into making this change.

Reply:
I looked into it a bit more and I think I prefer to leave the code as is because:
1. It is consistent with the other if checks above in this function
2. If I move the `if (!isWebGPUType(contextId))` earlier, then we won't hit the `ASSERT_NOT_REACHED()` later in this function if we fail to deal with a particular context type.